### ISSUE TITLE
[Product health] Monitor and address technical product metrics

### ISSUE THREAD

**Issue by eliserichards** [mozilla-mobile/fenix#9560]
[Engineering team OKRs](https://docs.google.com/document/d/18w_a5nSatAqL6TX6LinKsFY-93yU9u8E8cY8S5uCiiI/edit)

## Purpose
**Note:** this can be broken down as needed into separate cards. I kept it as one since that is how it's defined in the team's OKRs.

There are many ways to improve our monitoring of metrics that effect our app's perceived stability, such as:
- ability to easily monitor our crash rates and notify the team when there is a spike
   - how do we want sentry crashes to be tracked in github?
- defining criteria for each release
   - what do we want to be a part of each release?
   - define major, minor, crash release definitions
- working with the perf team to evaluate what their metrics are and how their priorities will effect the team's goals

## Acceptance Criteria
- [ ] ...

### App release criteria
- [ ] Define release criteria for new releases
- [ ] Clear process of what to monitor during app rollout

### Stability and crashes
- [ ] Define a process for monitoring Stability (filtering out native crashes)

### Performance team measurements
- [ ] Working w/ Performance team on their measurements

**Comment by liuche** [comment]
Work with @liuche to come up with outcomes for this sprint.

**Comment by liuche** [comment]
Notes from talking w/ Kate:
- Enforcing proper crash filing so it's easier to find in github
- Nightly: versions are hard to understand, maybe having a place to cross-reference?
- Nightly GPS / Sentry don't seem to show high crash rates
- GPS / Sentry don't seem to correlate well

Fenix 0.0a1 is Nightly in Crash-Stats
- To search Nightly, it's useful to filter by the last X days to get a good idea of Nightly state (bc it churns a lot)

**Comment by liuche** [comment]
Ended up talking with the GV team a bit in Matrix, and got more information about some GV bugs and their timelines.
- This is useful to request things like uplift for issues that are getting fixed in Nightly only
- Also got more stack traces, bc for some crash-stats stacktraces, you need to file a bugzilla bug (e.g. `ZygoteInit` bug had some hidden stacktraces)

**Comment by liuche** [comment]
E3 for breaking this down and coordinating with @kbrosnan

**Comment by liuche** [comment]
I added some crash stability user stories here: https://docs.google.com/document/d/1l5Yw3zfJQ41OpeqpwSHujZZ8TmmaAOeSy_iXiZvOnXs/edit#

I'll also summarize my current process as a release owner for release + rollout:
1) Go through QA-filed issues from release build testing
2) Once those issues are addressed, start rollout in Bugzilla
3) Each day, check the following places for crashes, and file + ping in-channel about new or high-incidence crashes:
    - Filter by release in Sentry to find new/high-incidence crashes
    - Check crash-stats for top crashers filtered by release version and day
    - Check Google Play Store dashboard for crashes per release, and cross-reference these Sentry (for reports + breadcrumbs) and crash-stats (for trends over time - this is difficult on Sentry)
4) Repeat from Step 2, and bump rollout percentage (1%, 20%, optional 50%, 100%)

**Comment by liuche** [comment]
@kglazko could you summarize what work you've done on ANRs here as well so this information is centralized? Optionally also add user stories to the stability user stories doc.

**Comment by sblatz** [comment]
@kbrosnan could I have your help in giving a brief overview of the current heads from each team and what they're responsibilities are?
